### PR TITLE
Enac student projects integration

### PR DESCRIPTION
**Pull request by gautierhattenberger**
Several results coming from student projects at Enac:
- update and fix for the JeVois smart cam driver
- improvement of the UWB-based localization, either with local or global positioning
- trajectory tracking based on shift around a reference
- some small corrections here and there

### CODE REVIEW

**Comment by noether** [comment]
Happy to review them, but it will take me some time. A lot of lines of code :P.

**Comment by noether** [comment]
@gautierhattenberger . Is it possible in the .xml (or somewhere else in the wiki) a link to the student reports linked with these modules? Even in French for me is okei :P. The idea is to follow better what are the different functionalities available in the module, and to understand faster the code eventually by others.

**Comment by gautierhattenberger** [comment]
the student reports are not that useful... but I have https://hal-enac.archives-ouvertes.fr/hal-01936955 which explains a part of the job done here. Not sure the wiki is the best place and I even removed some parts from the final version of my paper.

File: conf/flight_plans/rover_optitrack_PIR.xml [hunk 1]
```diff
@@ -0,0 +1,72 @@
+<!DOCTYPE flight_plan SYSTEM "flight_plan.dtd">
```
Comment (noether) [inline conf/flight_plans/rover_optitrack_PIR.xml]: What does it mean "PIR" ?

I suppose that in the near future we should have ./plans/flightplans ./plans/"groundplans"/ etc?

File: sw/airborne/filters/pid.h [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+ * @param pid pointer to PID structure
+ */
+static inline void reset_pid_f(struct PID_f *pid)
+{
+  pid->u = 0.f;
+  pid->e[0] = 0.f;
+  pid->e[1] = 0.f;
+  pid->sum = 0.f;
+}
+
```
Comment (noether) [inline sw/airborne/filters/pid.h]: I like this tool :P

What about a function for resetting only the integral part (sum)? Or better if you can set the value of "sum" to whatever value you wish, i.e., its initial condition. (I believe Simulink has it too).

File: conf/flight_plans/rover_optitrack_PIR.xml (hunk 1 above)
Comment (gautierhattenberger) [reply conf/flight_plans/rover_optitrack_PIR.xml]: who knows...
PIR stands for Projet Initiation Recherche. It is not really necessary to keep this one, but it has some nice moves for holonomic rover.

File: sw/airborne/filters/pid.h (hunk 2 above)
Comment (gautierhattenberger) [reply sw/airborne/filters/pid.h]: If you only need to reset/set the integral, maybe you can do it directly by changing the `sum` field ? Do you really think it is worth a dedicated function ? (probably better I guess...)

**Comment by gautierhattenberger** [comment]
I'll had the ref to one of module description

File: conf/flight_plans/rover_optitrack_PIR.xml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        <go from="P2" hmode="route" wp="P1" approaching_time="0"/>
+      </while>
+    </block>
+    <block name="circle">
+      <circle radius="nav.radius" wp="STDBY"/>
+    </block>
+    <block name="line_P1_P2_turn">
+      <go wp="P1" approaching_time="0"/>
+      <while cond="TRUE">
+        <go from="P1" hmode="route" wp="P2" approaching_time="0" pre_call="nav_set_heading_deg(turn_rate*block_time)"/>
```
Comment (noether) [inline conf/flight_plans/rover_optitrack_PIR.xml]: Out of curiosity. I am not familiar with pre_call inside the block. Is this called several times while in the "go"? What is the behavior of the rover here?

File: sw/airborne/filters/pid.h (hunk 2 above)
Comment (noether) [reply sw/airborne/filters/pid.h]: Not really. I think setting the field "sum" directly should be fine although not elegant :P

**Pull request merged**